### PR TITLE
Add mTLS encription between etcd and kube-apiserver in GCE

### LINKED ISSUE #70143: Lack of mTLS between kube-apiserver and etcd-client on GCE master
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug

/kind feature

**What happened**:
On GCE, the communication between etcd-client port and apiserver is NOT encrypted by mTLS by default. 

**What you expected to happen**:
Enable mTLS between etcd client and kube-apiserver on GCE.

**How to reproduce it (as minimally and precisely as possible)**:
cluster/kube-up.sh, kube-apiserver does not have "etcd-cafile", "etcd-certfile" or "etcd-keyfile" set, and etcd does not have "client-cert-auth", "trusted-ca-file", "cert-file" or "key-file" set. 

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.13
- Cloud provider or hardware configuration: GCE
- OS (e.g. from /etc/os-release): "Debian GNU/Linux"
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

/sig api-machinery
/sig gcp

### PR DESCRIPTION
**Pull request by wenjiaswe**
**What type of PR is this?**
> Uncomment only one, leave it on its own line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test

/kind feature

> /kind flake

**What this PR does / why we need it**:
This PR adds mTLS to secure communication between etcd-client port and apiserver. 
It generages CA cert and key, as well as corresponding server and client cert and key pairs, then uses them on the corresponding flags when start up apiserver and etcd as below.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #70143

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Enable mTLS encription between etcd and kube-apiserver in GCE
```

/sig api-machinery
/sig GCP

### CODE REVIEW

File: cluster/gce/gci/master-helper.sh
Comment (yguo0905) [inline cluster/gce/gci/master-helper.sh]: Changed by accident?

**Review by jpbetz — CHANGES_REQUESTED** [review]

File: cluster/gce/util.sh
Comment (gmarek) [inline cluster/gce/util.sh]: This doesn't seem to be used anywhere

**Comment by cjcullen** [comment]
high-level LGTM. I'll trust the others to dig into the bash :).

**Comment by wojtek-t** [comment]
@wenjiaswe - tests on higher scale didn't show any problems. So it's fine from performance/scalability POV (I didn't look into the code).

File: cluster/gce/util.sh
Comment (wenjiaswe) [reply cluster/gce/util.sh]: Thanks! I just removed that. Lesson learned, I should always check and make sure the comments are valid, especially copy comments from other functions...

**Comment by AishSundar** [comment]
Since we are just days before lifting code freeze, I feel its late in the release cycle to take in a feature when its not critical-urgent. Speaking with @wenjiaswe offline, we determined this will be a good candidate for 1.13.1 patch release.

/cc @aleksandra-malinowska @tpepper

**Comment by wenjiaswe** [comment]
@AishSundar Thanks Aish, let's do that.

**Comment by wenjiaswe** [comment]
/hold
check some setting

**Comment by wenjiaswe** [comment]
@gmarek @cjcullen  would you please take another look and see if you could help lgtm and approve? I addressed all the comments and rebased the PR.

File: cluster/gce/util.sh
Comment (gmarek) [inline cluster/gce/util.sh]: I don't think you want to allow those variables to be unset, as you're depending on them a line below it.

Comment (wenjiaswe) [reply cluster/gce/util.sh]: @gmarek I think it would be fine, as in [generate-etcd-cert](https://github.com/kubernetes/kubernetes/blob/7f23a743e8c23ac6489340bbb34fa6f1d392db9d/cluster/common.sh#L347), "If GEN_ETCD_CA_CERT or GEN_ETCD_CA_KEY is not specified, it will generates certs for CA." The existing [create-etcd-certs](https://github.com/kubernetes/kubernetes/blob/e6413d20427cb78c20e3e3d963682a9413244db3/cluster/gce/util.sh#L2069) is set this way too. Is it OK?

**Comment by tpepper** [comment]
Can folks comment on whether issue #70143 is being viewed as a feature or a bug?  And then whether this PR is going to be intended to be a cherry pick candidate to prior branches?  I ask because this PR is labeled feature, the issue is not labeled but has textual description as both bug and feature, and patch release managers were brought in early on cherry picks which were later closed.  It feels a bit up in the air right now...

**Comment by wenjiaswe** [comment]
@tpepper I am sorry for the confusion. This is a feature. And #70143 was labeled as feature (uncommented /kind feature) but somehow the markdown didn't show up right. I corrected that one. I am not going to cherry pick this back to previous versions. Thanks!

**Pull request merged**
